ddp-workspace/projects/ddp-sdk/src/lib/guards/irb.guard.ts

```diff
@@ -1,25 +1,32 @@
-import { Injectable } from '@angular/core';
+import { Injectable, Inject } from '@angular/core';
 import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
 import { IrbPasswordService } from '../services/irbPassword.service';
+import { ConfigurationService } from '../services/configuration.service';
 import { SessionMementoService } from '../services/sessionMemento.service';
 import { Observable, of } from 'rxjs';
-import { map, tap } from 'rxjs/operators';
+import { map, tap, catchError } from 'rxjs/operators';
 
 @Injectable()
 export class IrbGuard implements CanActivate {
     constructor(
         private session: SessionMementoService,
         private irbPassword: IrbPasswordService,
-        private router: Router) {
+        private router: Router,
+        @Inject('ddp.config') private config: ConfigurationService) {
     }
 
     public canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
         if (this.session.isAuthenticatedSession() || this.irbPassword.isIrbAuthenticated()) {
             return of(true);
         } else {
             return this.irbPassword.requiresIrbAuthentication().pipe(
-                tap(passwordRequired => passwordRequired && this.router.navigate(['/password'])),
-                map(passwordRequired => !passwordRequired));
+                tap(passwordRequired => passwordRequired && this.router.navigateByUrl(this.config.passwordPageUrl)),
+                map(passwordRequired => !passwordRequired),
+                catchError(() => {
+                    this.router.navigateByUrl(this.config.errorPageUrl);
+                    return of(false);
+                })
```

Comment:
If backend returns errors, a user will get stuck on this page:
![image](https://user-images.githubusercontent.com/30126907/102804014-36fdf580-43ca-11eb-9f85-1210b71860cc.png)
Let's redirect him to the error page.